lib/tree.rb

```diff
@@ -1,64 +1,152 @@
 class TreeNode
   attr_reader :key, :value
   attr_accessor :left, :right
-
-   def initialize(key, val)
+  
+  def initialize(key, val)
     @key = key
     @value = val
     @left = nil
     @right = nil
-   end
+  end
 end
 
 class Tree
   attr_reader :root
   def initialize
     @root = nil
   end
-
-  # Time Complexity: 
-  # Space Complexity: 
+  
+  # Time Complexity: O(log n)
+  # Space Complexity: O(1)
   def add(key, value)
```

Comment:
The method works, but your space complexity is off since you have to account for the callstack.  If the tree is balanced it's O(log n), and if unbalanced it's O(n).

Reply:
Hi I've updated the space complexity for the add, find and height methods that were incorrect. Could you take a look at those again and see if I'm on the right track?